src/sisyphus/cloud.clj

```diff
@@ -269,3 +277,52 @@
             relative-path (.substring remote-key prefix-length)
             local-path (join-path [path relative-path])]
         (download-blob! blob local-path)))))
+
+(defn create-compute-service
+  "Create an instance of com.google.api.services.compute.Compute to make requests with."
+  []
+  (let [transport (GoogleNetHttpTransport/newTrustedTransport)
+        factory (JacksonFactory/getDefaultInstance)
+        auth (into-array ["https://www.googleapis.com/auth/cloud-platform"])
+        credential (GoogleCredential/getApplicationDefault)
+        credential (if (.createScopedRequired credential)
+                     (.createScoped credential (Arrays/asList auth))
+                     credential)
+        builder (Compute$Builder. transport factory credential)]
+    (.setApplicationName builder "Gaia/0.0.1")
+    (.build builder)))
+
+(defn render-filter
+  "Render a map of options into the weird format that the compute instances api expects"
+  [instance-filter]
+  (string/join
+   " "
+   (map
+    (fn [[k v]]
+      (format
+       "(%s = %s)"
+       (name k)
+       (if (string? v)
+         (format "\"%s\"" v)
+         v)))
+    instance-filter)))
```

Comment:
Q. Does the `=` operator need to become `:` or something for one kind of match and `~` or something for a regex pattern? In the API explorer, I had very limited success making filters work. The default string match is supposed to be a word match, IIRC, but in the API explorer it seemed to match prefix strings.

Q. Would it be fitting to call a value-formatter like whatever `pr` calls rather than special-casing string?

Reply:
> Q. Does the = operator need to become : or something for one kind of match and ~ or something for a regex pattern?

Okay so this is pretty frustrating, it turns out the syntax for filters on the command line and in the HTTP API is different from the one in this Java API: https://developers.google.com/resources/api-libraries/documentation/compute/v1/java/latest/com/google/api/services/compute/Compute.Instances.List.html#setFilter-java.lang.String-

Specifically, this one only allows `=, !=, >, <` and requires everything to be in parens, whereas the command line one has support for partial matching and is a more straightforward comma separated list. No idea where this asymmetry came from but I think it was a bad decision. 

> Q. Would it be fitting to call a value-formatter like whatever pr calls rather than special-casing string?

Perhaps, I'll try that.

Reply:
Replaced conditional with `(with-out-str (pr v))`

Reply:
`with-out-str` -- So _that's_ how it's done! If only the docs for `pr` et al mentioned it.

Puzzling out the filter syntax is very frustrating. _If_ the client libraries call the same HTTP API that the command line tool and web API explorer all call, then there must be a way to express the same filter rules, even if the syntax differs. But it didn't work in my trials.

Reply:
The `gcloud` help message sort of suggests that it does regex filter in processing the server's response.

In the API explorer, `name=zoo*` lists `zookeeper-prime`, while `name=*prime` lists no servers. Fortunately our Sisyphus servers match `name=sisyphus-*`.